modules/model_package/spleen_segmentation/configs/metadata.json

```diff
@@ -0,0 +1,30 @@
+{
+    "version": "0.1.0",
+    "changelog": {
+        "0.1.0": "complete the model package",
+        "0.0.1": "initialize the model package structure"
+    },
+    "monai_version": "0.8.0",
+    "pytorch_version": "1.10.0",
+    "numpy_version": "1.21.2",
+    "optional_packages_version": {"nibabel": "3.2.1"},
+    "task": "Decathlon spleen segmentation",
+    "description": "A pre-trained model for volumetric (3D) segmentation of the spleen from CT image",
+    "authorship": "MONAI team",
+    "data_source": "Task09_Spleen.tar from http://medicaldecathlon.com/",
+    "dataset_dir": "/workspace/data/Task09_Spleen",
+    "image_classes": "single channel data, intensity scaled to [0, 1]",
```

Comment:
These tags are describing the input and output formats, that is shape and scaling. I would have this broken out into explicit tags so that this information can be used to construct the appropriate transforms to format inputs and posttransforms for outputs. Something like this adds more specific information:

```json
"data_format":{
    "inputs":[  
        {
            "name": "image",
            "type": "image",
            "format": "magniture",
            "shape": [1, 160, 160, 160],
            "type": "float32",
            "value_scale": [0,1],
            "center_crop": true,
        },
    ],
    "outputs":[
        {
            "name": "pred",
            "type": "image",
            "format": "segmentation",
            "from_network": {
                "shape": [2, 160, 160, 160],
                "type": "float32",
            },
            "shape": [1, 160, 160, 160],
            "type": "uint8",
            "classes": { 0: "background", 1: "spleen"},
            "argmax": true
        },
    ],
    "patch_inference": false
}
```

This provides information on what the input to the network is expected to be, that is a single float32 (1,160,160,160) image with values scaled to [0,1], so inferring what transforms to use to get this from arbitrary inputs can be done. For the output we have one image with an initial format having 2 channels which argmax is applied to to get the final 1 channel segmentation image, again using this information the transforms to do this can be inferred. 

Other information can be added to indicate how the network is to be applied to inputs, so "patch_inference" indicates the network is fed the full input image rather than being applied to patches of shape (1,160,160,160) from the input image.

Reply:
Interesting idea! I will try to enhance the metadata JSON.
I think it's a long way to go to finalize a perfect metadata dictionary, so for the first version, we should try to keep it simple and easy to use. Then we can add more and more items to it based on users' feedback, and also develop the `schemes` to format expected content.

Thanks.

Reply:
@ericspod , and Ming also provided more information in: https://github.com/Project-MONAI/MONAI/issues/3482#issuecomment-1018245931.

Thanks.

Reply:
Hi @ericspod ,

Thanks for your suggestion, I updated the `metadata.json` in this PR according to your suggestion.
I slightly changed several things in your proposal:
1. Changed the section name from `data_format` to `network_data_format`, as it actually only defines the input data for the network, not the data source for transforms, Dataset or DataLoader, etc.
2. Changed the `list` in `inputs` and `outputs` to `dict`, I feel it's more clear to access the content by `keys`.
3. I didn't define the `postprocessing` logic (like `argmax`, etc.) in the first version, as we also don't define the information for `preprocessing` here. I agree we can consider to add them in future, let's try to keep the metadata simple for the first version?

Could you please help take a look again?

Thanks in advance.